src/lfortran/semantics/ast_common_visitor.h

```diff
@@ -4061,6 +4061,20 @@ class CommonVisitor : public AST::BaseVisitor<Derived> {
                             a_value = ASRUtils::EXPR((ASR::make_IntegerConstant_t(al, loc,
                                                     a, int_type)));
                         }
+                        else if ( ASR::is_a<ASR::IntegerUnaryMinus_t>(*args[0].m_left) ) {
+                            ASR::expr_t* value = ASRUtils::expr_value(args[0].m_left);
+                            if ( value ) {
+                                int64_t a = ASR::down_cast<ASR::IntegerConstant_t>(value)->m_n - offset;
+                                a_value = ASRUtils::EXPR((ASR::make_IntegerConstant_t(al, loc,
+                                                    a, int_type)));
+                            }
+                        }
+
+                        if ( a_value != nullptr ) {
+                            int64_t a = ASR::down_cast<ASR::IntegerConstant_t>(a_value)->m_n;
+                            if ( a < 0 ) throw SemanticError("The first index in string section is less than 1", loc);
```

Comment:
```suggestion
                            if ( a < -1 ) throw SemanticError("The first index in string section is less than 0", loc);
```

This should take care of it.

Reply:
I think , for 0 start index is not valid in fortran and so error must be thrown for start index <= 0 or < 1 and so both the error message and condition are correct . I also checked the error message on my system and they are working well . So , I don't think that we have to change anything.